docs/kb/edk2-capsule-updates.md

```diff
@@ -0,0 +1,302 @@
+- one or more firmware update images (payloads)
+- metadata describing each of the payloads
+
+Creation of a capsule requires providing most of that information, the rest
+have defaults suitable for most use cases without modifications.
+
+### Required payload metadata
+
+#### Basic information
+
+- `Payload` — path to file containing payload data
+- `Guid` — GUID that identifies firmware variant
+- `FwVersion` — version of the firmware in the capsule
+- `LowestSupportedVersion` — the lowest version that new firmware will allow
+  to downgrade to
+
+Versions are 32-bit unsigned integers, requiring a scheme to translate
+something like v0.9.1 or v24.05.00.01 into an integer.  Different firmware
+variants can employ different translation schemes.
+
+#### Signing information
+
+Signing takes three keys (root, its subkey and subkey's subkey for signing):
+
+- `OpenSslTrustedPublicCertFile` — root key (CA) used for verification
+- `OpenSslOtherPublicCertFile` — subkey for capsule updates (included along
+  with signature)
+- `OpenSslSignerPrivateCertFile` — certificate for signing of the capsules
+
+This structure of signing keys is presupposed by `GenerateCapsule` and
+`Pkcs7Sign` but it seems to be a design decision of the tooling rather than a
+strict requirement on the length of a signing key chain by UEFI or cryptographic
+implementation in EDK2.
+
+### Optional payload metadata
+
+There are a few optional fields used to support updating multiple instances of
+the same device (`HardwareInstance`), firmware spread out across several
+flash chips (`UpdateImageIndex`), firmware with dependencies (`Dependencies`)
+as well as for strengthening security (`MonotonicCount`):
+
+- `HardwareInstance` (defaults to `0`)
+- `UpdateImageIndex` (defaults to `1`)
+- `Dependencies` (defaults to no dependencies) which is a string in Depex format
+- `MonotonicCount` (defaults to `0`) which is a 64-bit unsigned integer that is
+  supposed to guard against replay attacks (its value is appended to the payload
+  image before signing)
+
+## Capsule creation
+
+### Preparation
+
+#### Picking a GUID
+
+Every firmware variant should have its own GUID which appears in several places:
+
+- ESRT entry describing the firmware
+- FMP instance capable of updating the firmware
+- capsule metadata
+
+Switching between firmware variables is possible by adjusting GUID and version
+fields of a capsule meant to perform the switch.
+
+#### Preparing signing keys
+
+`BaseTools/Source/Python/Pkcs7Sign` contains everything necessary for testing
+purposes and that's what will be used below.
+
+Real-world usage requires proper key management.  Key generation is covered by
+the README of `Pkcs7Sign` as well as in [tianocore's wiki][wiki-keygen].
+`Pkcs7Sign` additionally describes how to integrate generated keys into EDK2's
+build system.
+
+[wiki-keygen]: https://github.com/tianocore/tianocore.github.io/wiki/Capsule-Based-System-Firmware-Update-Generate-Keys
```

Comment:
This mostly covers https://github.com/Dasharo/dasharo-issues/issues/806, we will only need to add user documentation there.